frontend/model/contracts/chatroom.js

```diff
@@ -136,6 +141,66 @@ function emitMessageEvent ({ contractID, hash }: {
   sbp('okTurtles.events/emit', `${CHATROOM_MESSAGE_ACTION}-${contractID}`, { hash })
 }
 
+export function makeMentionFromUsername (username: string): {
+  me: string, all: string
+} {
+  return {
+    me: `@${username}`,
+    all: '@all'
+  }
+}
+
+function addMention ({ contractID, messageId, datetime, text, username, chatRoomName }: {
+  contractID: string,
+  messageId: string,
+  datetime: string,
+  text: string,
+  username: string,
+  chatRoomName: string
+}): void {
+  /**
+   * If 'READY_TO_JOIN_CHATROOM' is false, it means not syncing chatroom
+  */
+  if (sbp('okTurtles.data/get', 'READY_TO_JOIN_CHATROOM')) {
+    return
+  }
+
+  sbp('state/vuex/commit', 'addChatRoomUnreadMention', {
+    chatRoomId: contractID,
+    messageId,
+    createdDate: new Date(datetime).getTime()
+  })
+
+  const rootGetters = sbp('state/vuex/getters')
+  const groupID = rootGetters.groupIdFromChatRoomId(contractID)
+  makeNotification({
+    title: `# ${chatRoomName}`,
+    body: text,
+    icon: rootGetters.globalProfile2(groupID, username).picture
+  })
+
+  sbp('okTurtles.events/emit', MESSAGE_RECEIVE)
```

Comment:
At the moment this is used by only one listener that doesn't care what the message is (it just plays a sound), and so it doesn't need any additional parameters. However, it's possible some other listener might be interested in knowing specifically which message was received. In that case it might be useful to pass in the `messageId` along with the `chatRoomId`.

Reply:
Is that over-engineering at this point?  If it is needed, we can add them.
But we don't need it at this time, do we?